entity-service/src/test/kotlin/com/egm/stellio/entity/repository/Neo4jRepositoryTests.kt

```diff
@@ -1149,6 +1150,185 @@ class Neo4jRepositoryTests : WithNeo4jContainer {
+                "typeNames" to setOf("https://ontology.eglobalmark.com/apic#Beehive")
+            ),
+            mapOf(
+                "attribute" to "temperature",
+                "typeNames" to setOf("https://ontology.eglobalmark.com/apic#Beehive")
+            ),
+            mapOf(
+                "attribute" to "HAS_VALUE",
+                "typeNames" to setOf("https://ontology.eglobalmark.com/apic#Sensor")
+            ),
+            mapOf(
+                "attribute" to "deviceParameter",
+                "typeNames" to setOf("https://ontology.eglobalmark.com/apic#Sensor")
+            ),
+            mapOf(
+                "attribute" to "https://uri.etsi.org/ngsi-ld/location",
+                "typeNames" to setOf("https://ontology.eglobalmark.com/apic#Sensor")
+            ),
+            mapOf(
+                "attribute" to "isContainedIn",
+                "typeNames" to setOf("https://ontology.eglobalmark.com/apic#Sensor")
+            )
+        )
+        assertEquals("Got the following types instead: $attribute", 7, attribute.size)
+        assertArrayEquals(arrayOf(expectedAttributes), arrayOf(attribute))
+    }
+
+    @Test
+    fun `it should return an emptyMap for unknown attributeType`() {
+        createEntity(
+            "urn:ngsi-ld:Beehive:TESTC".toUri(),
+            listOf("https://ontology.eglobalmark.com/apic#Beehive"),
+            mutableListOf(
+                Property(name = "temperature", value = 36),
+                Property(name = "humidity", value = 65)
+            )
+        )
+
+        val attributesInformation = neo4jRepository
+            .getAttributeInformation("unKnown")
+
+        assertTrue(attributesInformation.isEmpty())
+    }
+
+    @Test
+    fun `it should retrieve attributes information for two entities if attribute is a relationship type`() {
+        createEntity(
+            "urn:ngsi-ld:Beehive:TESTC".toUri(),
+            listOf("https://ontology.eglobalmark.com/apic#Beehive"),
+            mutableListOf(
+                Property(name = "temperature", value = 36),
+                Property(name = "humidity", value = 65)
+            )
+        )
+        val secondEntity = createEntity(
+            "urn:ngsi-ld:Beehive:TESTB".toUri(),
+            listOf("https://ontology.eglobalmark.com/apic#Beehive"),
+            mutableListOf(
+                Property(name = "temperature", value = 30),
+                Property(name = "humidity", value = 61)
+            ),
+            NgsiLdGeoPropertyInstance.toWktFormat(GeoPropertyType.Point, listOf(24.30623, 60.07966))
+        )
+        createRelationship(EntitySubjectNode(secondEntity.id), "observedBy", partialTargetEntityUri)
+
+        val actualAttributesInformation = neo4jRepository
+            .getAttributeInformation("observedBy")
+
+        val expectedAttributesInformation = mapOf(
+            "attributeName" to "observedBy",
+            "attributeTypes" to listOf("Relationship", "GeoProperty"),
```

Comment:
`GeoProperty`??!

Reply:
if an entity has location as a property, GeoProperty tag would be added in attributeTypes list.

Reply:
in this specific case, `observedBy` is not a geo-property, so `GeoProperty` should not be in the list